internal/ethapi/api.go

```diff
@@ -1431,16 +1432,26 @@ func newRPCPendingTransaction(tx *types.Transaction, current *types.Header, conf
 }
 
 // newRPCTransactionFromBlockIndex returns a transaction that will serialize to the RPC representation.
-func newRPCTransactionFromBlockIndex(b *types.Block, index uint64, config *params.ChainConfig, backend Backend) *RPCTransaction {
+func newRPCTransactionFromBlockIndex(b *types.Block, index uint64, config *params.ChainConfig, db ethdb.Database) *RPCTransaction {
 	txs := b.Transactions()
-	log.Error(fmt.Sprintf("I am here", len(txs), "index", 0))
-	tx, _, _, _ := rawdb.ReadBorTransaction(backend.ChainDb(), b.Hash())
-	if tx != nil {
-		txs = append(txs, tx)
+
+	borReceipt := rawdb.ReadBorReceipt(db, b.Hash(), b.NumberU64())
+	if borReceipt == nil {
+		return nil
 	}
+
+	tx, _, _, _ := rawdb.ReadBorTransaction(db, borReceipt.TxHash)
+
+	if tx == nil {
+		return nil
+	}
+
+	txs = append(txs, tx)
```

Comment:
Is this slice is sorted after the operation? If not, how could we get by index?

Reply:
The index of state-sync-transaction is always the last one in the block. That's why I believe it is safe to append it to the end of the list.

Reply:
Got it. Should work, however could you add a small test to document expected behaviour?